pkg/controller/account/account_controller.go

```diff
@@ -169,6 +166,25 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
+		awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, setupClientInput)
+		if err != nil {
+			reqLogger.Error(err, "failed building operator AWS client")
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				// If it's AccessDenied we want to just delete the finalizer and continue as we assume
+				// the credentials have been deleted by the customer
+				case "AccessDenied":
+					err = r.removeFinalizer(currentAcctInstance, awsv1alpha1.AccountFinalizer)
+					if err != nil {
+						reqLogger.Error(err, "failed removing account finalizer")
+						return reconcile.Result{}, err
+					}
+					return reconcile.Result{}, nil
```

Comment:
In here (or before this code block), I think we'll want to add a check to only remove the finalizer for non-CCS accounts for safety.

For some accounting, purposes, we may want to add a Prometheus counter and/or log message of when this code path is reached, broken down by Account type (CCS vs non-CCS) so we can look back and see some data about the ratio. (At this time I think a log message will suffice as part of a "break/fix" solution and we can revisit the Prometheus metric later.)